src/main/resources/com/google/api/codegen/py/main.snip

```diff
@@ -211,24 +212,52 @@
                 :func:`google.gax.construct_settings` for the structure of
                 this data. Falls back to the default config if not specified
                 or the specified config is missing data points.
-              app_name (string): The codename of the calling service.
-              app_version (string): The version of the calling service.
+              app_name (string): The name of the application calling
+                the service. Recommended for analytics purposes.
+              app_version (string): The version of the application calling
+                the service. Recommended for analytics purposes.
+              lib_name (string): The API library software used for calling
+                the service. (Unless you are writing an API client itself,
+                leave this as default.)
+              lib_version (string): The API library software version used
+                for calling the service. (Unless you are writing an API client
+                itself, leave this as default.)
+              metrics_headers (dict): A dictionary of values for tracking
```

Comment:
SGTM assuming @bjwatson is OK with this.

(I'm not sure what "semi-private" means given that any developer will be able to see these arguments in the public docs, and this constructor will be the entry point to the client library for any API that doesn't have a handwritten layer.)